docs/docs/concepts/memory.md

```diff
@@ -132,44 +155,124 @@ def summarize_conversation(state: State):
+
+This has some benefits. It's a completely separate process from your application, so generally incurs no latency. It also splits up the application logic from the memory logic, making it more modular and easy to manage.
+
+This also has several downsides. It may not happen realtime, so users will not immediately see memory updated. You also have to think more about when to trigger this job - how do you know a conversation is finished?
+## Update own instructions
+
+This is an example of long term memory.
+
+"Reflection" or "Meta-prompting" steps can use an LLM to generate or refine its own prompts or instructions. This approach allows the system to dynamically update and improve its own behavior, potentially leading to better performance on various tasks. This is particularly useful for tasks where the instructions are challenging to specify a priori. 
 
 Meta-prompting can use past information to update the prompt. As an example, this [Tweet generator](https://www.youtube.com/watch?v=Vn8A3BxfplE) uses meta-prompting to iteratively improve the summarization prompt used to generate high quality paper summaries for Twitter. In this case, we used a LangSmith dataset to house several papers that we wanted to summarize, generated summaries using a naive summarization prompt, manually reviewed the summaries, captured feedback from human review using the LangSmith Annotation Queue, and passed this feedback to a chat model to re-generate the summarization prompt. The process was repeated in a loop until the summaries met our criteria in human review.
 
-## Retrieving relevant information from long-term storage
+This will utilize the memory store concept above to store the updated instructions in a shared namespace. This namespace will have only a single item (unless you want to update instructions specific for each user, but that's a separate issue). This will look something like:
+
+```python
+# Node that *uses* the instructions
+def call_model(state: State, store: BaseStore):
+    instructions = store.search(("instructions",))[0]
+    # Application logic
+    prompt = prompt_template.format(instructions=instructions.value["instructions"])
+    ...
+
+
+# Node that updates instructions
+def update_instructions(state: State, store: BaseStore):
+    current_instructions = store.search(("instructions",))[0]
+    # Memory logic
+    prompt = prompt_template.format(instructions=instructions.value["instructions"], conversation=state["messages"])
+    output = llm.invoke(prompt)
+    new_instructions = output['new_instructions']
+    store.put(("instructions",), current_instructions.key, {"instructions": new_instructions})
+    ...
+```
+
+![](img/memory/update-instructions.png)
+### Remember a profile
```

Comment:
nit: Remember vs. learn? (for foreign speakers remember and recall can be synonymous)